components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/validators/LogoutReqSignatureValidator.java

```diff
@@ -0,0 +1,256 @@
+                        StandardCharsets.UTF_8.name()));
+            }
+            throw new SecurityException("Couldn't extract the signature value from the query string parameter: "
+                    + signatureQueryParam);
+        } catch (UnsupportedEncodingException e) {
+            throw new IdentityException("Error occurred while decoding signature query parameter: "
+                    + signatureQueryParam, e);
+        }
+    }
+
+    /**
+     * Extract the signed content string from the query string in the request.
+     *
+     * @param queryString SAML request (passed an an HTTP query parameter).
+     * @return byte[]     Signed content.
+     * @throws SecurityException If process of constructing signed content fails.
+     */
+    private static byte[] getSignedContent(String queryString) throws SecurityException {
+
+        String sigendContent = buildSignedContentString(queryString);
+        if (StringUtils.isEmpty(sigendContent)) {
+            String message = "Couldn't extract signed content string from query string: " + queryString;
+            if (log.isDebugEnabled()) {
+                log.debug(message);
+            }
+            throw new SecurityException(message);
+        }
+        return sigendContent.getBytes(StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Extract the raw request parameters and build a string representation of
+     * the content that was signed.
+     *
+     * @param queryString SAML request (passed an an HTTP query parameter).
+     * @return String     Representation of the signed content.
+     * @throws SecurityException thrown if there is an error during request processing.
+     */
+    private static String buildSignedContentString(String queryString) throws SecurityException {
+
+        StringBuilder builder = new StringBuilder();
+
+        if (StringUtils.isBlank(URISupport.getRawQueryStringParameter(queryString,
+                HTTP_POST_PARAM_SAML2_AUTH_REQ))) {
+            throw new SecurityException("Process of extracting SAMLRequest from query string failed: "
+                    + queryString);
+        }
+        appendParameter(builder, queryString, HTTP_POST_PARAM_SAML2_AUTH_REQ);
+        // This is optional.
+        appendParameter(builder, queryString, RELAY_STATE);
+        // This is mandatory, but has already been checked in superclass.
+        appendParameter(builder, queryString, SIGNATURE_ALGORITHM);
+
+        return builder.toString();
+    }
+
+    /**
+     * Append raw query string parameter it to the string builder.
+     *
+     * @param builder     String builder to which to append the parameter
+     * @param queryString The URL query string containing parameters
+     * @param paramName   The name of the parameter to append
+     * @return true       If raw string of parameter is not null.
+     */
+    private static boolean appendParameter(StringBuilder builder, String queryString,
```

Comment:
why is this method having the return type as boolean? seems like the return value is not used in the places where this method is getting called.

Reply:
Fixed with PR #133